stripe-core/src/main/java/com/stripe/android/core/model/parsers/StripeErrorJsonParser.kt

```diff
@@ -19,7 +19,13 @@ class StripeErrorJsonParser : ModelJsonParser<StripeError> {
                     message = optString(errorObject, FIELD_MESSAGE),
                     param = optString(errorObject, FIELD_PARAM),
                     type = optString(errorObject, FIELD_TYPE),
-                    docUrl = optString(errorObject, FIELD_DOC_URL)
+                    docUrl = optString(errorObject, FIELD_DOC_URL),
+                    extraFields = errorObject
+                        .optJSONObject(FIELD_EXTRA_FIELDS)?.let { extraFieldsJson ->
+                            extraFieldsJson.keys().asSequence()
+                                .map { key -> key to extraFieldsJson.get(key).toString() }
```

Comment:
Looking [here](https://livegrep.corp.stripe.com/view/stripe-internal/pay-server/lib/api_errors/data/error.rb#L48) it appears that this can be a bool, int, or string. Does this line always convert to a string and if so, should it preserve type?

Reply:
Yeah, these always parses as string. Since the list of extra fields is dynamic, we cannot figure which of the 3 types we should expect, so the best we can do is generify parsing as string (added a test to verify these [3 types are properly mapped](https://github.com/stripe/stripe-android/pull/5278/files#diff-4e07600a016e9f41cd17000deb6761db2acf2801861b5b39b1bc5d4dfcbb6bd1)).

field consumers that know what type to expect can always convert the field on their end.

Reply:
Alternatively we could have a polymorphic type, but seems like an overkill for this:

```
sealed interface ExtraField {
  class Bool(val value: Boolean): ExtraField
  class Number(val value: Int): Extrafield
  ...
}
```